NitroxClient/GameLogic/PlayerManager.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NitroxClient.GameLogic
+{
+    public class PlayerManager
+    {
+        private readonly Dictionary<string, RemotePlayer> playersById = new Dictionary<string, RemotePlayer>();
+
+        public RemotePlayer this[string playerId]
+        {
+            get
+            {
+                return FindPlayer(playerId);
+            }
+            set
+            {
+                if (value == null)
+                    RemovePlayer(playerId);
+                else
+                    playersById[playerId] = value;
+            }
+        }
+
+        public RemotePlayer FindPlayer(string playerId, bool createPlayer = false)
```

Comment:
Optional RemotePlayer to encapsulate null OR always instantiate it if it is not found.  I don't think there is really a good reason for a null reference to be floating around.